pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authorization/AuthorizationService.java

```diff
@@ -70,17 +90,38 @@ public AuthorizationService(ServiceConfiguration conf, ConfigurationCacheService
         }
     }
 
+    private static <T> CompletableFuture<T> anyMatch(List<? extends CompletionStage<? extends T>> l,
```

Comment:
`anyMatch` doesn't seem to be right to me when multiple authentication providers are applied to. Is it better to check them in sequence?

Reply:
@sijie considering https://github.com/apache/pulsar/pull/6661#issuecomment-608543559, maybe we should just define the authz provider to use on the auth provider, so dynamically use the right authz provider depending on the auth provider.

Reply:
So put `Set<String> getAuthZMethodsNames();` on `AuthenticationProvider` and use the defined authZ provider depending on the auth provider in `AuthorizationService`. WDYT? @sijie @merlimat

Reply:
```
    /**
     * @return the authorization methods names supported by this provider
     */
    default Set<String> getAuthZMethodsNames() {
        return Sets.newHashSet("PulsarAuthorizationProvider");
    }
```

Reply:
The point is: how to get the right auth provider when using authZ provider methods :/.

Reply:
Put the auth provider method name to `AuthenticationDataSource`?

Reply:
Hum, or not. Anyway, I'm waiting for your point of view.

Reply:
Or just a simple way: as it's done for auth providers, just verify there is at least one authZ provider that authorizes the request this is why the `anyMatch`.